From: Michal Hocko <mhocko@suse.cz>
Subject: hugetlb, kabi: do not account hugetlb pages as NR_FILE_PAGES
Patch-mainline: never
References: bnc#930092

hugetlb.h header will pull in struct mempolicy definition transitively and
this confuses kABI checker which now complains:
/usr/src/packages/BUILD/kernel-default-3.0.101/linux-3.0/mm/filemap.c:182: warning: delete_from_page_cache: modversion changed because of changes in struct mempolicy (became defined)
and later
KABI: symbol read_cache_page_gfp(mm/filemap/built-in) changed crc from 0x2654d727 to 0xf2d98303

Fix this by hiding the include from the kABI checker.

Signed-off-by: Michal Hocko <mhocko@suse.cz>

---
 mm/filemap.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/mm/filemap.c
+++ b/mm/filemap.c
@@ -34,7 +34,9 @@
 #include <linux/hardirq.h> /* for BUG_ON(!in_atomic()) only */
 #include <linux/memcontrol.h>
 #include <linux/cleancache.h>
+#ifndef __GENKSYMS__
 #include <linux/hugetlb.h>
+#endif
 #include "internal.h"
 
 /*
